nuxeo-coldstorage/src/test/java/org/nuxeo/coldstorage/auth/ColdStorageS3TestHelper.java

```diff
@@ -0,0 +1,163 @@
+
+    protected static final String AWS_REGION_ENV_VAR = "COLDSTORAGE_AWS_REGION";
+
+    protected final AWSCredentialsProvider awsCredentialsProvider = new DefaultAWSCredentialsProviderChain();
+
+    protected AWSCredentials credentials;
+
+    protected String mainBucket;
+
+    protected String glacierBucket;
+
+    protected String region;
+
+    protected String bucketPrefix;
+
+    public static ColdStorageS3TestHelper getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new ColdStorageS3TestHelper();
+        }
+        return INSTANCE;
+    }
+
+    private ColdStorageS3TestHelper() {
+        this.credentials = awsCredentialsProvider.getCredentials();
+
+        log.info("ColdStorage integration tests will be done using AWS");
+
+        this.mainBucket = Objects.requireNonNull(getEnvValue(AWS_MAIN_BUCKET_NAME_ENV_VAR),
+                "Main bucket name is required");
+        this.glacierBucket = Objects.requireNonNull(getEnvValue(AWS_GLACIER_BUCKET_NAME_ENV_VAR),
+                "Glacier bucket name is required");
+        this.bucketPrefix = Objects.requireNonNull(getEnvValue(AWS_BUCKET_PREFIX_ENV_VAR), "Bucket prefix is required");
+        this.region = Objects.requireNonNull(getEnvValue(AWS_REGION_ENV_VAR), "Region is required");
+    }
+
+    public AWSCredentials getCredentials() {
+        return credentials;
+    }
+
+    public void cleanMainBucket(String blobName) {
+        AmazonS3 amazonS3 = getAmazonS3(blobName);
+        ObjectListing s3Objects = amazonS3.listObjects(this.getMainBucket(), this.getBucketPrefix());
+        s3Objects.getObjectSummaries()
+                 .forEach(s3Object -> amazonS3.deleteObject(this.getMainBucket(), s3Object.getKey()));
+    }
+
+    public void cleanGlacierBucket(String blobName) {
+        AmazonS3 amazonS3 = getAmazonS3(blobName);
+        ObjectListing s3Objects = amazonS3.listObjects(this.getGlacierBucket(), this.getBucketPrefix());
+        s3Objects.getObjectSummaries()
+                 .forEach(s3Object -> amazonS3.deleteObject(this.getGlacierBucket(), s3Object.getKey()));
+    }
+
+    public AmazonS3 getAmazonS3(String provider) {
+        S3BlobProvider  s3BlobProvider = (S3BlobProvider ) Framework.getService(BlobManager.class)
+                                                                  .getBlobProvider(provider);
+        return s3BlobProvider.config.amazonS3;
+    }
+
+    public String getMainBucket() {
+        return mainBucket;
+    }
+
+    public String getGlacierBucket() {
+        return glacierBucket;
+    }
+
+    public String getRegion() {
+        return region;
+    }
+
+    public String getBucketPrefix() {
+        return bucketPrefix;
+    }
+
+    protected static String getEnvValue(String key) {
+        return StringUtils.trim(System.getenv(key));
+    }
+
+    public  static void updateObjectClassStorage(DocumentModel document) {
```

Comment:
Even it's a helper class. It will be helpful to have a java documentation where you explain why we need to do manually. 

Also I think you should retrieve and check his existence before the copy/move to be more robut. If in future we avoid using the S3 Life cycle.